index.d.ts

```diff
@@ -100,6 +100,14 @@ declare module '@primer/components' {
     setOpen: (open: boolean) => void
   }
 
+  type SetTimeout = (handler: TimerHandler, timeout?: number, ...args: any[]) => number
+  type ClearTimeout = (id: number) => void
+
+  export const useSafeTimeout: () => {
+    safeSetTimeout: SetTimeout,
+    safeClearTimeout: ClearTimeout
+  }
```

Comment:
Can we avoid writing out the types for SetTImeout and ClearTimeout?

```suggestion
  export const useSafeTimeout: () => {
    safeSetTimeout: typeof window.setTimeout,
    safeClearTimeout: typeof window.clearTimeout
  }
```